scene/main/canvas_layer.cpp

```diff
@@ -87,6 +87,17 @@ Transform2D CanvasLayer::get_transform() const {
 	return transform;
 }
 
+Transform2D CanvasLayer::get_final_transform() const {
+	Transform2D follow;
+	if (is_following_viewport()) {
+		follow.scale(Vector2(get_follow_viewport_scale(), get_follow_viewport_scale()));
+		if (vp) {
+			follow = vp->get_canvas_transform() * follow;
+		}
+	}
+	return follow * transform;
```

Comment:
The transforms multiplication should probably be avoided when not following the viewport.

Reply:
Makes sense to be more performant. I used this notation, because I found it in other places of the source code, like https://github.com/godotengine/godot/blob/be0b77fd29b07ec9c05f7e532733a5aa2dda0346/scene/main/viewport.cpp#L4112-L4124

Reply:
Fun fact: [you're the author of the linked code](https://github.com/godotengine/godot/commit/ffe42171a52ad83a20e7c048cf92bcc84251c669). :slightly_smiling_face:

I mean it's probably not a big deal but I'd say let's not rely on the compiler to optimize it. For example here it's done separately in the branches:
https://github.com/godotengine/godot/blob/22a09fef5d56fc7c37d70118532509076ebd7b12/scene/main/canvas_item.cpp#L145-L153

I think a similar if-else in here would make sense too.

Reply:
You are right :-D But I am quite certain that I saw that format also in other places in the source code.